bigframes/operations/_matplotlib/core.py

```diff
@@ -28,3 +29,44 @@ def draw(self) -> None:
     @property
     def result(self):
         return self.axes
+
+
+class SamplingPlot(MPLPlot):
+    @abc.abstractproperty
+    def _kind(self):
+        pass
+
+    def __init__(self, data, **kwargs) -> None:
+        self.kwargs = kwargs
+        self.data = self._compute_plot_data(data)
+
+    def generate(self) -> None:
+        self.axes = self.data.plot(kind=self._kind, **self.kwargs)
+
+    def _compute_plot_data(self, data):
+        # TODO: Cache the sampling data in the PlotAccessor.
+        sampling_n = self.kwargs.pop("sampling_n", 100)
+        sampling_random_state = self.kwargs.pop("sampling_random_state", 0)
+        return (
+            data.sample(n=sampling_n, random_state=sampling_random_state)
+            .to_pandas()
+            .sort_index()
```

Comment:
why is the sort necessary?

Reply:
the index of `df.sample` is not sorted as https://screenshot.googleplex.com/9nhptZFVA8sZJCz